classes/importer.php

```diff
@@ -644,24 +622,21 @@ protected function _getupdatesfromldap($ldap, $ldaptimestamp = null) {
                                 }
                                 $result[$attr] = $email;
                             } else if (
-                                (core_text::strtolower('sn') == $attr)
-                                       || (core_text::strtolower('givenname') == $attr)
+                                        core_text::strtolower('sn') == $attr
+                                        || (core_text::strtolower('ucsfEduPreferredLastName') == $attr)
+                                        || (core_text::strtolower('givenname') == $attr)
+                                        || (core_text::strtolower('ucsfEduPreferredGivenName') == $attr)
+                                        || (core_text::strtolower('initials') == $attr)
+                                        || (core_text::strtolower('ucsfEduPreferredMiddleName') == $attr)
+                                        || (core_text::strtolower('displayName') == $attr)
                             ) {
-                                // Fixing: this field could have 'question mark' in it.
-                                // If so, just remove it.  ($DB->execute() does not like '?')
+                                // Fixing: these fields could have 'question mark' in it.
+                                // If so, just remove it.  ($DB->execute() does not like '?').
```

Comment:
turns out that's by design. for reference: see https://moodle.org/mod/forum/discuss.php?d=173498

my takeaway is that `$DB->execute()` should be avoided, if possible.

Reply:
Created #10 for this._